packages/data/src/world/mq/ice_cavern_mq.yml

```diff
@@ -2,42 +2,98 @@
   dungeon: IC
   exits:
     "Zora Fountain Frozen": "true"
-    "Ice Cavern Main": "has_ranged_weapon || has_explosives" #The pot carry is very trick territory; this crystal switch is a problem.
+    "Ice Cavern Room After Entrance Boulder": "true"
   locations:
     "MQ Ice Cavern Pot Entrance": "true"
-"Ice Cavern Main":
+"Ice Cavern Room After Entrance Boulder":
   dungeon: IC
   exits:
-    "Ice Cavern Map Room": "soul_wolfos && soul_freezard && (has_weapon || can_use_sticks || has_explosives)"
-    "Ice Cavern Compass Room": "is_adult && has_blue_fire"
-    "Ice Cavern Big Room": "has_blue_fire"
+    "Ice Cavern": "true"
+    "Ice Cavern Ledge After Shiek": "climb_anywhere || hookshot_anywhere"
+    "Ice Cavern After Initial Ice Block": "event(ICE_MQ_FIRST_CRYSTAL)"
+  events:
+    ICE_MQ_FIRST_CRYSTAL: "has_ranged_weapon || has_explosives || (is_adult && (has(SWORD_BIGGORON) || can_use_mask_blast))" #Biggoron Jump Slash as Adult hits it, so does Adult Blast Mask. The pot carry is very trick territory; this crystal switch is a problem.
+"Ice Cavern After Initial Ice Block":
+  dungeon: IC
+  exits:
+    "Ice Cavern Room After Entrance Boulder": "event(ICE_MQ_FIRST_CRYSTAL)"
+    "Ice Cavern Main Room": "true"
   locations:
     "MQ Ice Cavern Pot First Room 1": "true"
     "MQ Ice Cavern Pot First Room 2": "true"
```

Comment:
This doesn't seem right. aren't these pots in the first corridor, which is why you are even able to throw a pot at the crystal?

Reply:
No. The first corridor has one pot: "MQ Ice Cavern Pot Entrance"

These pots Nax labelled semi-incorrectly. THey are right after the ice block where you need to hit the crystal.